crates/pathfinder/src/storage/schema/revision_0014.rs

```diff
@@ -0,0 +1,396 @@
+
+    if todo == 0 {
+        return Ok(());
+    }
+
+    tracing::info!(
+        num_transactions=%todo,
+        "Decompressing and migrating transactions, this may take a while.",
+    );
+
+    let mut compressor = zstd::bulk::Compressor::new(1).context("Create zstd compressor")?;
+
+    let mut stmt = transaction
+        .prepare("SELECT hash, tx FROM starknet_transactions")
+        .context("Prepare transaction query")?;
+    let mut update_stmt = transaction
+        .prepare("UPDATE starknet_transactions SET tx = :tx WHERE hash = :hash")
+        .context("Prepare transaction update statement")?;
+    let mut query_class_hash_stmt = transaction
+        .prepare("SELECT hash FROM contracts WHERE address = :contract_address")
+        .context("Prepare class hash lookup statement")?;
+
+    let mut old_uncompressed_size = 0usize;
+    let mut old_compressed_size = 0usize;
+    let mut new_uncompressed_size = 0usize;
+    let mut new_compressed_size = 0usize;
+
+    let mut processed_rows = 0;
+    let batch_size = (todo / 11).max(100000);
+    let start_of_run = std::time::Instant::now();
+    let mut start_of_batch = start_of_run;
+
+    let mut rows = stmt.query([])?;
+    while let Some(r) = rows.next()? {
+        let transaction_hash = r.get_ref_unwrap("hash").as_blob()?;
+        let tx = r.get_ref_unwrap("tx").as_blob()?;
+
+        old_compressed_size += tx.len();
+        let tx = zstd::decode_all(tx).context("Decompressing transaction")?;
+        old_uncompressed_size += tx.len();
+        let tx: transaction::Transaction = serde_json::de::from_slice(&tx).context(format!(
+            "Deserializing transaction '{}'",
+            String::from_utf8_lossy(&tx)
+        ))?;
+
+        // Fix missing class_hash in deploy transactions
+        let tx = match tx {
+            transaction::Transaction::Deploy(mut deploy) => {
+                if deploy.class_hash.is_none() {
+                    let class_hash = query_class_hash_stmt.query_row(
+                        named_params![":contract_address": deploy.contract_address.0.as_be_bytes()],
+                        |r| {
+                            let class_hash = r.get_ref_unwrap("hash").as_blob().unwrap();
+                            let class_hash = StarkHash::from_be_slice(class_hash).unwrap();
+                            Ok(ClassHash(class_hash))
+                        },
+                    ).context("Query class hash for contract")?;
+                    deploy.class_hash = Some(class_hash);
+                    tracing::trace!(transaction_hash = ?deploy.transaction_hash, "Fixed missing class_hash for deploy transaction");
+                }
+                transaction::Transaction::Deploy(deploy)
+            }
+            transaction::Transaction::Declare(declare) => {
+                transaction::Transaction::Declare(declare)
+            }
+            transaction::Transaction::Invoke(invoke) => transaction::Transaction::Invoke(invoke),
```

Comment:
Could just continue on these, it will be less writes, and doing less writes is better.

It at least seems like it would be cheaper not to recompress.